x-pack/plugins/lens/public/shared_components/coloring/palette_configuration.tsx

```diff
@@ -0,0 +1,348 @@
+          display="rowCompressed"
+        >
+          <EuiButtonGroup
+            isFullWidth
+            legend={i18n.translate('xpack.lens.table.dynamicColoring.rangeType.label', {
+              defaultMessage: 'Value type',
+            })}
+            data-test-subj="lnsDatatable_dynamicColoring_custom_range_groups"
+            name="dynamicColoringRangeType"
+            buttonSize="compressed"
+            options={[
+              {
+                id: `${idPrefix}percent`,
+                label: i18n.translate('xpack.lens.table.dynamicColoring.rangeType.percent', {
+                  defaultMessage: 'Percent',
+                }),
+                'data-test-subj': 'lnsDatatable_dynamicColoring_rangeType_groups_percent',
+              },
+              {
+                id: `${idPrefix}number`,
+                label: i18n.translate('xpack.lens.table.dynamicColoring.rangeType.number', {
+                  defaultMessage: 'Number',
+                }),
+                'data-test-subj': 'lnsDatatable_dynamicColoring_rangeType_groups_number',
+              },
+            ]}
+            idSelected={
+              activePalette.params?.rangeType
+                ? `${idPrefix}${activePalette.params?.rangeType}`
+                : `${idPrefix}percent`
+            }
+            onChange={(id) => {
+              const newRangeType = id.replace(
+                idPrefix,
+                ''
+              ) as RequiredPaletteParamTypes['rangeType'];
+
+              const params: CustomPaletteParams = { rangeType: newRangeType };
+              if (isCurrentPaletteCustom) {
+                const { min: newMin, max: newMax } = getDataMinMax(newRangeType, dataBounds);
+                const { min: oldMin, max: oldMax } = getDataMinMax(
+                  activePalette.params?.rangeType,
+                  dataBounds
+                );
+                const newColorStops = roundStopValues(
+                  remapStopsByNewInterval(colorStopsToShow, {
+                    oldInterval: oldMax - oldMin,
+                    newInterval: newMax - newMin,
+                    newMin,
+                    oldMin,
+                  })
+                );
+                const stops = getPaletteStops(
+                  palettes,
+                  { ...activePalette.params, colorStops: newColorStops, ...params },
+                  { dataBounds }
+                );
+                params.colorStops = newColorStops;
+                params.stops = stops;
+                params.rangeMin = newColorStops[0].stop;
+                params.rangeMax = newColorStops[newColorStops.length - 1].stop;
+              } else {
+                params.stops = getPaletteStops(
+                  palettes,
+                  { ...activePalette.params, ...params },
+                  { prevPalette: activePalette.name, dataBounds }
+                );
+              }
+              setPalette(mergePaletteParams(activePalette, params));
```

Comment:
This code creates a buggy-feeling UI in the following case:

1. Select a default palette
2. Switch to exact value mode
3. The numbers aren't rounded at all

It looks like the rounding only happens when I have a custom palette, but maybe the right thing to do is apply rounding for presentation, but model using full precision. What do you think about this?